### PR TITLE
wazevo: lower atomic rmw to ssa

### PR DESCRIPTION
**Pull request by anuraaga**
@mathetake Back to unknown territory so let me know if I'm doing anything wrong. I'll finish testcases if this looks good for `RMWAnd`.

I referred to wasmtime a bit for ssa pointers, notably they have a similar rmwop enum

https://github.com/bytecodealliance/wasmtime/blob/0d662c9d3dd1db8490e30124c9126bfa00a9a5c6/cranelift/codegen/src/ir/atomic_rmw_op.rs#L10
https://github.com/bytecodealliance/wasmtime/blob/0d662c9d3dd1db8490e30124c9126bfa00a9a5c6/cranelift/wasm/src/code_translator.rs#L2996

### CODE REVIEW

**Comment by mathetake** [comment]
looks good to me

**Comment by anuraaga** [comment]
Thanks for checking @mathetake, I have filled the remaining tests

File: internal/engine/wazevo/ssa/instructions.go [hunk 1]
```diff
@@ -651,6 +651,41 @@ const (
...
+	AtomicRmwOpSub
+	// And
+	AtomicRmwOpAnd
+	// Or
+	AtomicRmwOpOr
+	// Xor
+	AtomicRmwOpXor
+	// Exchange
+	AtomicRmwOpXchg
+)
```
Comment (mathetake) [inline internal/engine/wazevo/ssa/instructions.go:669]: could you add proper comments

**Review by mathetake — APPROVED** [review]

File: internal/engine/wazevo/ssa/instructions.go (hunk 1 above)
Comment (anuraaga) [reply internal/engine/wazevo/ssa/instructions.go]: Yeah, updated

**Pull request merged**